### PR TITLE
keccak-hash: switch benches to criterion

### PR DESCRIPTION
**Pull request by ordian**
* keccak-hash: switch benches to criterion
* check benches on CI
* apply cargo fix
* updates docs to 2018 edition

### CODE REVIEW

File: primitive-types/impls/serde/src/serialize.rs [hunk 1]
```diff
@@ -78,7 +78,7 @@ pub enum FromHexError {
 impl std::error::Error for FromHexError {}
 
 impl fmt::Display for FromHexError {
-	fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+	fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
```
Comment (NikVolf) [inline primitive-types/impls/serde/src/serialize.rs]: seems redundant?

Comment (ordian) [reply primitive-types/impls/serde/src/serialize.rs]: this was generated by `cargo fix --edition-idioms` so I assumed it's part of the idioms lint, see also https://doc.rust-lang.org/edition-guide/rust-2018/ownership-and-lifetimes/the-anonymous-lifetime.html

but if you think this is useless, I can revert the change

Comment (NikVolf) [reply primitive-types/impls/serde/src/serialize.rs]: I don't see why we need useless generic argument, generated or not 🧐

File: keccak-hash/benches/keccak_256.rs [hunk 2]
```diff
@@ -14,37 +14,36 @@
...
+		b.iter(|| {
+			let _out = keccak(black_box(empty));
+		})
+	});
 }
 
-#[bench]
-fn bench_keccak_256_with_typical_input(b: &mut Bencher) {
+pub fn keccak_256_with_typical_input(c: &mut Criterion) {
 	let data: Vec<u8> = From::from("some medum length string with important information");
```
Comment (dvdplm) [inline keccak-hash/benches/keccak_256.rs]:
```suggestion
	let data: Vec<u8> = From::from("some medium length string with important information");
```

Comment (dvdplm) [reply keccak-hash/benches/keccak_256.rs]: 51 bytes… I have no idea what the median input size is, but it'd be nice to know.

File: kvdb-rocksdb/Cargo.toml [hunk 3]
```diff
@@ -27,6 +27,7 @@ parity-util-mem = { path = "../parity-util-mem", version = "0.4", default-featur
 [dev-dependencies]
 alloc_counter = "0.0.4"
 criterion = "0.3"
+ethereum-types = { path = "../ethereum-types" }
```
Comment (dvdplm) [inline kvdb-rocksdb/Cargo.toml]: Why was this needed?

Comment (ordian) [reply kvdb-rocksdb/Cargo.toml]: I broke this in https://github.com/paritytech/parity-common/pull/301/files#diff-3206254ae7bfdf4d978075eb4cafbdcfR30, but we didn't have CI checks to catch it because `keccak-hash` benches didn't compile on stable I think?

Comment (ordian) [reply kvdb-rocksdb/Cargo.toml]: (this is a bench-only dependency)
